src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../../contracts/core/PaymentCoordinator.sol";
+import "../../contracts/interfaces/IPaymentCoordinator.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+
+import "../mocks/ERC20Mock.sol";
+
+
+
+import "forge-std/Test.sol";
+
+
+contract PaymentCoordinatorTest is Test {
+    Vm cheats = Vm(HEVM_ADDRESS);
+    PaymentCoordinator public paymentCoordinator;
+    PaymentCoordinator public paymentCoordinatorImplementation;
+    ProxyAdmin public proxyAdmin;
+    ERC20Mock public dummyToken;
+    uint256 MAX_BIPS = 10000;
+    uint256 eigenLayerShareBIPs = 1000;
+    uint256 rootPostDelay = 7200;
+
+    event PaymentReceived(address indexed receivedFrom, PaymentCoordinator.Payment payment);
+    event NewMerkleRootPosted(PaymentCoordinator.MerkleRootPost merkleRootPost);
+
+
+    function setUp() public {
+        proxyAdmin = new ProxyAdmin();
+        paymentCoordinatorImplementation = new PaymentCoordinator();
+        paymentCoordinator = PaymentCoordinator(
+                                address(new TransparentUpgradeableProxy(
+                                    address(paymentCoordinatorImplementation), 
+                                    address(proxyAdmin),
+                                    abi.encodeWithSelector(
+                                        paymentCoordinatorImplementation.initialize.selector,
+                                        address(this),
+                                        address(this),
+                                        eigenLayerShareBIPs,
+                                        rootPostDelay
+                                    )
+                                )));
+        
+        dummyToken = new ERC20Mock();
+    }
+
+    function testInitialize() public {
+        require(paymentCoordinator.rootPublisher() == address(this), "rootPublisher should be set");
+        require(paymentCoordinator.merkleRootActivationDelay() == 7200, "merkleRootActivationDelay should be set");
+        require(paymentCoordinator.eigenLayerShareBIPs() == 1000, "eigenLayerShareBIPs should be set");
+        require(paymentCoordinator.owner() == address(this), "owner should be set");
+    }
+
+    function testMakePayment(uint256[] memory amounts) public {
+        cheats.assume(amounts.length > 0);
+        uint256 sum = _sum(amounts);
+        PaymentCoordinator.Payment memory payment;
+        payment.token = dummyToken;
+        payment.amounts = amounts;
+
+        uint256 balanceBefore = dummyToken.balanceOf(address(paymentCoordinator));
+        uint256 cumumlativeEarningsBefore = paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken);
+        cheats.expectEmit(address(paymentCoordinator));
```

Comment:
I'm pretty sure using `expectEmit` like this purely checks that the address emits _some_ event? 
Please correct me if I'm wrong but I've been using `expectEmit(true, true, true, true, address(...))` because as I understand it that's the strictest check.
Documentation here https://book.getfoundry.sh/cheatcodes/expect-emit

Reply:
no I believe expectEmit checks that the provided addres emits the specified event below it.  The `expectEmit(true, true, true, true, address(...))` variation of the statement is for checking emissions of multiple events

Reply:
chatted about this offline, foundry docs seem unclear, `expectEmit(true, true, true, true, address(...))` seems safest/strictest.